lark/lexer.py

```diff
@@ -468,6 +469,25 @@ def __copy__(self):
     _Token = Token
 
 
+class PostLexThread(LexerThread):
+    def __init__(self, lexer: 'Lexer', lexer_state: LexerState, postlex: 'PostLex'):
+        super().__init__(lexer, lexer_state)
+        self.postlex = postlex
+
+    @classmethod
+    def from_text(cls, lexer: 'Lexer', text_or_slice: TextOrSlice, postlex: 'PostLex') -> 'PostLexThread':
+        text = TextSlice.cast_from(text_or_slice)
+        return cls(lexer, LexerState(text), postlex)
+
+    def lex(self, parser_state):
+        # Get tokens from the underlying lexer and process with postlex
+        tokens = super().lex(parser_state)
+        return self.postlex.process(tokens)
+
+    def __copy__(self):
+        return type(self)(self.lexer, copy(self.state), self.postlex)
```

Comment:
Are we sure that the postlexer shouldn't be copied too?

Reply:
But it does hold a state, e.g. https://github.com/lark-parser/lark/blob/master/lark/indenter.py#L33

Reply:
Although, tbh, those shouldn't be instance attributes, but local variables inside of `process`. I think this postlexer design should already be broken on the current main if this copy method is called since `lexer=PostLexConnector` doesn't get copied either.

Reply:
i see, i agree those don't really make sense as instance attributes. it makes sense that a postlexer instance should be agnostic to a stream of tokens. this matter is probably beyond the scope of this PR though.